datafusion/physical-expr/src/aggregate/utils.rs

```diff
@@ -21,28 +21,13 @@ use arrow::array::ArrayRef;
 use datafusion_common::{Result, ScalarValue};
 use datafusion_expr::Accumulator;
 
-/// Extract scalar values from an accumulator. This can return an error if the accumulator
-/// has any non-scalar values.
-pub fn get_accum_scalar_values(accum: &dyn Accumulator) -> Result<Vec<ScalarValue>> {
-    accum
-        .state()?
-        .iter()
-        .map(|agg| agg.as_scalar().map(|v| v.clone()))
-        .collect::<Result<Vec<_>>>()
-}
-
-/// Convert scalar values from an accumulator into arrays. This can return an error if the
-/// accumulator has any non-scalar values.
+/// Convert scalar values from an accumulator into arrays.
 pub fn get_accum_scalar_values_as_arrays(
     accum: &dyn Accumulator,
 ) -> Result<Vec<ArrayRef>> {
     accum
         .state()?
         .iter()
-        .map(|v| {
-            v.as_scalar()
-                .map(|s| vec![s.clone()])
-                .and_then(ScalarValue::iter_to_array)
-        })
+        .map(|s| ScalarValue::iter_to_array(vec![s.clone()]))
```

Comment:
Just an observation, this doesn't need to be a vec?

Reply:
Good call -- changed to 

```rust
        .map(|s| s.to_array_of_size(1))
```

in 1165cffe7